### PR TITLE
ERA-811: Remove hardcoded party IDs

### PR DESCRIPTION
**Pull request by gaborh-da**

### CODE REVIEW

File: sandbox.Dockerfile [hunk 1]
```diff
@@ -7,8 +7,13 @@ ARG sdk_vsn=1.1.1
 
 FROM digitalasset/daml-sdk:${sdk_vsn} AS source
 
+USER root
```
Comment (tamaskalcza-da) [inline sandbox.Dockerfile:10]: Normally we would not need this. I think it's worth to add a comment or a TODO that would indicate when we remove this.

File: sandbox.Dockerfile [hunk 2]
```diff
@@ -7,8 +7,13 @@ ARG sdk_vsn=1.1.1
...
+
+RUN echo 'hosts: files dns' > /etc/nsswitch.conf
+
 WORKDIR /home/daml/
 
+USER daml
```
Comment (tamaskalcza-da) [inline sandbox.Dockerfile:16]: Normally we would not need this. I think it's worth to add a comment or a TODO that would indicate when we remove this.

File: daml/Triggers/AcknowledgeAndDiscloseTrigger.daml [hunk 3]
```diff
@@ -125,10 +121,18 @@ appointmentTest = scenario do
       originalPolicy = referralScenario.originalPolicy
       disclosureList = [provider1, provider2]
 
+  dRule <- patient1 `submit`
```
Comment (tamaskalcza-da) [inline daml/Triggers/AcknowledgeAndDiscloseTrigger.daml:124]:
```suggestion
  disclosureRule <- patient1 `submit`
```
Applies to other parts of the code.

File: daml/Triggers/AcknowledgeAndDiscloseTrigger.daml [hunk 4]
```diff
@@ -22,90 +22,86 @@ import DA.List.Total
...
-    whenSome selectedPolicyOpt $ \selectedPolicy ->
-      dedupExercise npCid $ AcknowledgeAndDisclose with
-        policyCid = selectedPolicy
-        receivers = receivers
+acknowledgeAndDiscloseRule : Party -> ACS -> Time -> Maps.Map CommandId [Command] -> () -> TriggerA ()
+acknowledgeAndDiscloseRule _party activeContracts _ _ () =
+  whenSome dRuleOpt $
+    \(_, dRule) ->
+      forA_ notifyPatientContracts $ \(npCid, notifyPatient) -> do
+        let selectedPolicyOpt =
```
Comment (tamaskalcza-da) [inline daml/Triggers/AcknowledgeAndDiscloseTrigger.daml:43]: I don't like this Hungarian notation: "someValue**Opt**".

File: daml/Triggers/EvaluateReferralTrigger.daml [hunk 5]
```diff
@@ -29,11 +30,10 @@ evaluateReferralTrigger = Trigger
...
-    (networkContract, _) =
-      headNote
-        "Every provider should have exactly one network contract"
-        (getContracts @ProviderNetworkContract activeContracts)
+  let requests = map fst $ getContracts @ReferralRequest activeContracts
+      networkContractOpt =
+        head $ getContracts @ProviderNetworkContract activeContracts
 
-  forA_ requests (`dedupExercise` EvaluateReferral networkContract)
+  whenSome networkContractOpt $
```
[the remark this answers is not in the document]
Comment (gaborh-da) [reply daml/Triggers/EvaluateReferralTrigger.daml:37]: There should be either one or none. I deliberately used the optional specific approach.

File: daml/Triggers/AcknowledgeAndDiscloseTrigger.daml (hunk 4 above)
Comment (gaborh-da) [reply daml/Triggers/AcknowledgeAndDiscloseTrigger.daml:43]: I changed it but from some point of view it is still Hungarian notation. I need to differentiate between this and then actual value when there is some.

File: daml/Triggers/AcknowledgeAndDiscloseTrigger.daml (hunk 3 above)
Comment (gaborh-da) [reply daml/Triggers/AcknowledgeAndDiscloseTrigger.daml:124]: Okay

File: sandbox.Dockerfile [hunk 6]
```diff
@@ -7,8 +7,14 @@ ARG sdk_vsn=1.1.1
 
 FROM digitalasset/daml-sdk:${sdk_vsn} AS source
 
+USER root
+
+# TODO: Remove this when SDK's Docker container gets fixed.
```
Comment (tamaskalcza-da) [inline sandbox.Dockerfile:12]: Id there’s a DAML issue, please link it. If not, please create one.

Comment (gaborh-da) [reply sandbox.Dockerfile:12]: The issue was reported weeks ago, I don't see anything in the tracker. I will ask the DAML team and add something to their list and to here.
Please do the same thing for the original occurence of this fix:
https://github.com/digital-asset/ex-bond-issuance/blob/master/sandbox.Dockerfile#L12

We don't have to wait for that with this PR.

File: daml/DemoOnboardScenario/StartScript.daml [hunk 7]
```diff
@@ -122,4 +122,9 @@ insurancePoliciesSetSingle = do
     patient1 `submit`
         exerciseCmd insurancePolicy DisclosePolicy with receivers = [provider2]
 
-    return ()
+    patient1 `submit`
+        createCmd DisclosureRule with
+            operator = operator
+            owner = patient1
+            receivers = [provider1, provider2]
+    debug "Ledger initialized."
```
Comment (tamaskalcza-da) [inline daml/DemoOnboardScenario/StartScript.daml:130]: _Minor_: I think it's worth to log at the beginning of the script as well.

**Pull request merged**